linker/shared/src/main/scala/org/scalajs/linker/analyzer/Infos.scala

```diff
@@ -727,6 +730,12 @@ object Infos {
             case VarDef(_, _, vtpe, _, _) =>
               builder.maybeAddReferencedClass(vtpe)
 
+            case prop: LinkTimeProperty =>
+              if (!linkTimeProperties.validate(prop)) {
+                throw new LinkingException(
+                    s"Couldn't find a link-time property '${prop.name}' of type ${prop.tpe}.")
+              }
```

Comment:
Here we validate the existence of LInkTimeProperty, and throw an exception when it's invalid.
However, it seems like we should store the error somehow in 

https://github.com/scala-js/scala-js/blob/03e6b54943d2ce490c4be7c8f4f65fa21411d6e5/linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala#L122 instead of throwing an error?

Reply:
Yes, indeed. All "user-level" link-time errors should be reported in the `_errors` of the `Analyzer`.

This could be done by adding a field to the `ReachabilityInfo` to remember errors that need to be reported when the `Analyzer` starts looking into a reachable method.